tests/test_compute_everything.py

```diff
@@ -0,0 +1,238 @@
+        "desc.geometry.curve.SplineXYZCurve": {"grid": curvegrid1},
+        "desc.magnetic_fields._core.OmnigenousField": {"grid": fieldgrid},
+    }
+
+    with open("tests/inputs/master_compute_data_rpz.pkl", "rb") as file:
+        master_data_rpz = pickle.load(file)
+    with open("tests/inputs/master_compute_data_xyz.pkl", "rb") as file:
+        master_data_xyz = pickle.load(file)
+    this_branch_data_rpz = {}
+    this_branch_data_xyz = {}
+    update_master_data_rpz = False
+    update_master_data_xyz = False
+    error_rpz = False
+    error_xyz = False
+
+    # some things can't compute "phi" and therefore can't convert to XYZ basis
+    no_xyz_things = ["desc.magnetic_fields._core.OmnigenousField"]
+
+    with warnings.catch_warnings():
+        # Max resolution of master_compute_data.pkl limited by GitHub file
+        # size cap at 100 mb, so can't hit suggested resolution for some things.
+        warnings.filterwarnings("ignore", category=ResolutionWarning)
+        for p in things:
+            names = {
+                name
+                for name in data_index[p]
+                # Skip these quantities as they should be covered in other tests.
+                if not data_index[p][name]["source_grid_requirement"]
+            }
+            this_branch_data_rpz[p] = things[p].compute(
+                list(names), **grid.get(p, {}), basis="rpz"
+            )
+            # make sure we can compute everything
+            assert this_branch_data_rpz[p].keys() == names, (
+                f"Parameterization: {p}. Can't compute "
+                + f"{names - this_branch_data_rpz[p].keys()}."
+            )
+            # compare data against master branch
+            error_rpz, update_master_data_rpz = _compare_against_master(
+                p,
+                this_branch_data_rpz,
+                master_data_rpz,
+                error_rpz,
+                update_master_data_rpz,
+            )
+
+            # test compute in XYZ basis
+            if p in no_xyz_things:
+                continue
+            # remove quantities that are not implemented in the XYZ basis
+            # TODO: generalize this instead of hard-coding for "grad(B)" & dependencies
+            names_xyz = (
+                names - {"grad(B)", "|grad(B)|", "L_grad(B)"}
+                if "grad(B)" in names
+                else names
+            )
+
+            this_branch_data_xyz[p] = things[p].compute(
+                list(names_xyz), **grid.get(p, {}), basis="xyz"
+            )
+            assert this_branch_data_xyz[p].keys() == names_xyz, (
+                f"Parameterization: {p}. Can't compute "
+                + f"{names_xyz - this_branch_data_xyz[p].keys()}."
+            )
+            error_xyz, update_master_data_xyz = _compare_against_master(
+                p,
+                this_branch_data_xyz,
+                master_data_xyz,
+                error_xyz,
+                update_master_data_xyz,
```

Comment:
my suggestion would be to just compute in `rpz` and `xyz` and check that `convert(xyz_data to rpz) == rpz_data`. Then just check if `rpz_data == master_rpz_data` .

Reply:
I like this